vslib/src/sai_vs_switch_MLNX2700.cpp

```diff
@@ -350,111 +350,121 @@ static sai_status_t create_default_trap_group()
 {
-    SWSS_LOG_ENTER();
+    // 8 in and 8 out queues per port
+    const uint32_t port_qos_queues_count = 16;
+    std::vector<sai_object_id_t> queues;
 
-    SWSS_LOG_INFO("create qos queues");
+    for (uint32_t i = 0; i < port_qos_queues_count; ++i)
+    {
+        sai_object_id_t queue_id;
 
-    sai_object_id_t switch_id = ss->getSwitchId();
+        sai_attribute_t attr[2];
 
-    // 8 in and 8 out queues per port
-    const uint32_t port_qos_queues_count = 16;
+        attr[0].id = SAI_QUEUE_ATTR_INDEX;
+        attr[0].value.u8 = (uint8_t)i;
+        attr[1].id = SAI_QUEUE_ATTR_PORT;
+        attr[1].value.oid = port_id;
 
-    std::vector<sai_object_id_t> copy = port_list;
+        CHECK_STATUS(vs_generic_create(SAI_OBJECT_TYPE_QUEUE, &queue_id, switch_id, 2, attr));
 
-    copy.push_back(cpu_port_id);
+        queues.push_back(queue_id);
+    }
 
-    for (auto &port_id : copy)
-    {
-        std::vector<sai_object_id_t> queues;
+    sai_attribute_t attr;
 
-        for (uint32_t i = 0; i < port_qos_queues_count; ++i)
-        {
-            sai_object_id_t queue_id;
+    attr.id = SAI_PORT_ATTR_QOS_NUMBER_OF_QUEUES;
+    attr.value.u32 = port_qos_queues_count;
 
-            sai_attribute_t attr[2];
+    CHECK_STATUS(vs_generic_set(SAI_OBJECT_TYPE_PORT, port_id, &attr));
 
-            attr[0].id = SAI_QUEUE_ATTR_INDEX;
-            attr[0].value.u8 = (uint8_t)i;
+    attr.id = SAI_PORT_ATTR_QOS_QUEUE_LIST;
+    attr.value.objlist.count = port_qos_queues_count;
+    attr.value.objlist.list = queues.data();
 
-            attr[1].id = SAI_QUEUE_ATTR_PORT;
-            attr[1].value.oid = port_id;
+    CHECK_STATUS(vs_generic_set(SAI_OBJECT_TYPE_PORT, port_id, &attr));
 
-            CHECK_STATUS(vs_generic_create(SAI_OBJECT_TYPE_QUEUE, &queue_id, switch_id, 2, attr));
+    return SAI_STATUS_SUCCESS;
+}
 
-            queues.push_back(queue_id);
-        }
+static sai_status_t create_qos_queues()
+{
+    SWSS_LOG_ENTER();
 
-        sai_attribute_t attr;
+    SWSS_LOG_INFO("create qos queues");
 
-        attr.id = SAI_PORT_ATTR_QOS_NUMBER_OF_QUEUES;
-        attr.value.u32 = port_qos_queues_count;
+    sai_object_id_t switch_id = ss->getSwitchId();
 
-        CHECK_STATUS(vs_generic_set(SAI_OBJECT_TYPE_PORT, port_id, &attr));
+    std::vector<sai_object_id_t> copy = port_list;
 
-        attr.id = SAI_PORT_ATTR_QOS_QUEUE_LIST;
-        attr.value.objlist.count = port_qos_queues_count;
-        attr.value.objlist.list = queues.data();
+    copy.push_back(cpu_port_id);
 
-        CHECK_STATUS(vs_generic_set(SAI_OBJECT_TYPE_PORT, port_id, &attr));
+    for (auto &port_id : copy)
+    {
+        create_qos_queues_per_port(switch_id, port_id);
     }
 
     return SAI_STATUS_SUCCESS;
 }
 
-static sai_status_t create_ingress_priority_groups()
+static sai_status_t create_ingress_priority_groups_per_port(sai_object_id_t switch_id, sai_object_id_t port_id)
```

Comment:
we keep 1 param per line and adding "_in_" "_out"_ prefixes, please keep formating